### PR TITLE
README: Add a section on how to determine the touchscreen parameters

### PR DESCRIPTION
**Pull request by jwrdegoede**
Add a section on how to correctly determine the touchscreen parameters to help users of new device models find the right parameters.

Note the part about testing the parameters depends on the new `i2c_touchscreen_props=` kernel commandline argument which is supported in kernel 6.10 and newer.

### CODE REVIEW

**Comment by jwrdegoede** [comment]
Note this assumes that this kernel patch will get merged as a "fix" for 6.10 :
https://lore.kernel.org/platform-driver-x86/20240523143601.47555-1-hdegoede@redhat.com/

**Comment by hmtheboy154** [comment]
thank you so much @jwrdegoede & @onitake. We can finally moved on from gslx680-acpi & getting data directly from BlissOS

**Comment by onitake** [comment]
Thank you for taking the time to do this @jwrdegoede .

I think we can also start cleaning up the the parts about gslx680_acpi. I wouldn't completely delete them, but maybe move them over to the https://github.com/onitake/gslx680-acpi/ repository and only keep the historic files and device README entries as a reference. I'd also remove the parts from the main README and replace them with a link to gslx680-acpi instead.

This doesn't need to happen now, though. Give me a bit of time to review your PR, and then we can merge it. I'll look into moving the old documentation later.

File: README.md
Comment (onitake) [inline README.md:200]:
```suggestion
and rebuilt [(*)](kernel-recompile-debian.md) to test parameter changes.
```

Comment (onitake) [reply README.md:200]: I think you should also replace the `(*)` with a link description. Clicking `(*)` is hard on a small screen and has other accessibility issues. Something like `[see kernel-recompile-debian.md for Debian-specific instructions]`, or something shorter.

File: README.md
Comment (onitake) [inline README.md:212]:
```suggestion
not work. In this case you may still be able to download a factory OS image from
the device vendor's website, but not all vendors offer this.
```

File: README.md
Comment (onitake) [inline README.md:216]:
```suggestion
silead_ts, which should already be enabled in your Linux distribution's kernel, will print
```

File: README.md
Comment (onitake) [inline README.md:247]:
```suggestion
makes proper contact with the screen. If that doesn't help, then you have
```

File: README.md
Comment (onitake) [inline README.md:296]: I'd move this part after the next paragraph. You already mentioned how to detect swapped axes, so adding the correct flags should be straight-forward.

File: README.md
Comment (onitake) [inline README.md:319]: Do we need to mention how to determine the number of multi-touch points?
I believe we've seen devices with 5 and some with 10.

**Review by onitake — CHANGES_REQUESTED** [review]

File: README.md
Comment (jwrdegoede) [reply README.md:296]: > I'd move this part after the next paragraph. You already mentioned how to detect swapped axes, so adding the correct flags should be straight-forward.

I did the split of first testing with only min/max settings vs adding swapping deliberately. The text to test the axis talks about top/bottom and left/right edges. Depending on form factor users are likely to define which edges are top/bottom and what is left/right from e.g. holding the device in landscape orientation. But the actual LCD panel could be a portrait panel.

So even though the user swiped from top to bottom when testing the X-axis, which feels like swapping is needed, swapping might not be necessary (and actually would be wrong to do) because they were holding the LCD panel 90° rotated from its native panel orientation.

So I prefer to keep this as is and first have users test with only min/max values set.

File: README.md
Comment (jwrdegoede) [reply README.md:319]: Basically all but the very oldest Silead touchscreens only found on really cheap Allwinner A13 / A23 tablets can do 10 points. So I plan to submit a patch upstream to make 10 the default and drop all the boilerplate PROPERTY_ENTRY_U32("silead,max-fingers", 10) code from drivers/platform/x86/touchscreen_dmi.c I will submit a patch series for this upstream soon.

**Comment by jwrdegoede** [comment]
@onitake thank you for the review. I have just pushed a new version which incorporates all your suggestions and changes the `[(*)]` for the kernel-recompile-debian.md link to `[(see e.g. kernel-recompile-debian.md)]` .

I also have a green light for adding the kernel patch to 6.10, so I will push this out to pdx86/fixes sometime today and then send out a pull-request to Linus sometime next week.

**Comment by jwrdegoede** [comment]
For my last comment s/today/Monday/ for pushing out a new fixes branch, since I need to wait for 6.10-r1 to get tagged first.

**Comment by jwrdegoede** [comment]
> I think we can also start cleaning up the the parts about gslx680_acpi. I wouldn't completely delete them, but maybe move them over to the https://github.com/onitake/gslx680-acpi/ repository and only keep the historic files and device README entries as a reference. I'd also remove the parts from the main README and replace them with a link to gslx680-acpi instead.
> 
> This doesn't need to happen now, though. Give me a bit of time to review your PR, and then we can merge it. I'll look into moving the old documentation later.

Moving the gslx68-acpi info out of the main README into a separate document sounds good to me, thank you.

**Review by onitake — APPROVED** [review]
LGTM, thanks!

**Comment by jwrdegoede** [comment]
Thank you for the approval, I'll merge this when I send the pull-request with the kernel patch to Linus.

**Pull request merged**
